scheduler/test/cook/test/scheduler/scheduler.clj

```diff
@@ -661,19 +723,46 @@
                                                  #mesomatic.types.Resource{:name "gpus", :type :value-scalar :scalar 2.0 :role "*"}],
                                      :attributes [],
                                      :executor-ids []}
-        adapter (sched/->VirtualMachineLeaseAdapter offer when)]
+        adapter (sched/->VirtualMachineLeaseAdapter offer now)]
 
     (is (= (.getId adapter) "my-offer-id"))
     (is (= (.cpuCores adapter) 40.0))
     (is (= (.diskMB adapter) 6000.0))
-    (is (= (.getOfferedTime adapter) when))
+    (is (= (.getOfferedTime adapter) now))
     (is (= (.getVMID adapter) "my-slave-id"))
     (is (= (.hostname adapter) "slave3"))
     (is (= (.memoryMB adapter) 5000.0))
     (is (= (.getScalarValues adapter) {"gpus" 2.0 "cpus" 40.0 "disk" 6000.0 "mem" 5000.0 "ports" 0.0}))
     (is (= (-> adapter .portRanges first .getBeg) 31000))
     (is (= (-> adapter .portRanges first .getEnd) 32000))))
 
+(deftest test-virtual-machine-lease-adapter-k8s
+  ;; ensure that the VirtualMachineLeaseAdapter can successfully handle an offer from Kubernetes.
+  (let [now (System/currentTimeMillis)
+        offer {:id  {:value "my-offer-id"}
+               :framework-id "my-framework-id"
+               :slave-id {:value "my-slave-id"},
+               :hostname "slave3",
+               :resources [{:name "cpus", :type :value-scalar, :scalar 40.0, :ranges [], :set #{}, :role "*"}
+                           {:name "mem", :type :value-scalar, :scalar 5000.0, :ranges [], :set #{}, :role "*"}
+                           {:name "disk", :type :value-scalar, :scalar 6000.0, :ranges [], :set #{}, :role "*"}
+                           {:name "ports", :type :value-ranges, :scalar 0.0, :ranges [{:begin 31000, :end 32000}], :set #{}, :role "*"}
+                           {:name "gpus", :type :value-text->scalar :text->scalar {"nvidia-tesla-p100" 2} :role "*"}],
+               :attributes [],
+               :executor-ids []}
+        adapter (sched/->VirtualMachineLeaseAdapter offer now)]
+
+    (is (= (.getId adapter) "my-offer-id"))
+    (is (= (.cpuCores adapter) 40.0))
+    (is (= (.diskMB adapter) 6000.0))
+    (is (= (.getOfferedTime adapter) now))
+    (is (= (.getVMID adapter) "my-slave-id"))
+    (is (= (.hostname adapter) "slave3"))
+    (is (= (.memoryMB adapter) 5000.0))
+    (is (= (.getScalarValues adapter) {"cpus" 40.0 "disk" 6000.0 "mem" 5000.0 "ports" 0.0}))
```

Comment:
Should we test the GPU fields?

Reply:
yes, added testing lines to check that gpu field in attribute map is as expected